### DISCUSSION TITLE
Should SPARQL loading not also support formats with named graphs?

### DISCUSSION THREAD

**Discussion by aucampia** [RDFLib/rdflib#2411] [TO-DO list]
Currently SPARQL queries seem to only be able to load sources that are in formats that don't support named graphs:

https://github.com/RDFLib/rdflib/blob/ad56044cf3033f8d2a3c625d322f3d091d7c9322/rdflib/plugins/sparql/sparql.py#L314-L333

I'm not sure if this is right.

**Comment by edmondchuc** [comment]
i'm guessing this load function is only applicable to SPARQL CONSTRUCT queries? 

According to the spec https://www.w3.org/TR/sparql11-query/#construct 

> The `CONSTRUCT` query form returns a single RDF graph specified by a graph template.
